tensorflow_addons/layers/polynomial.py

```diff
@@ -0,0 +1,162 @@
+        A tuple of 2 (batch_size, `input_dim`) dimensional inputs.
+
+    Output shape:
+        A single (batch_size, `input_dim`) dimensional output.
+    """
+
+    @typechecked
+    def __init__(
+        self,
+        projection_dim: int = None,
+        use_bias: bool = True,
+        kernel_initializer: types.Initializer = "truncated_normal",
+        bias_initializer: types.Initializer = "zeros",
+        kernel_regularizer: types.Regularizer = None,
+        bias_regularizer: types.Regularizer = None,
+        **kwargs,
+    ):
+        super(PolynomialCrossing, self).__init__(**kwargs)
+
+        self.projection_dim = projection_dim
+        self.use_bias = use_bias
+        self.kernel_initializer = tf.keras.initializers.get(kernel_initializer)
+        self.bias_initializer = tf.keras.initializers.get(bias_initializer)
+        self.kernel_regularizer = tf.keras.regularizers.get(kernel_regularizer)
+        self.bias_regularizer = tf.keras.regularizers.get(bias_regularizer)
+
+        self.supports_masking = True
+
+    def build(self, input_shape):
+        if not isinstance(input_shape, (tuple, list)) or len(input_shape) != 2:
+            raise ValueError(
+                "Input shapes must be a tuple or list of size 2, "
+                "got {}".format(input_shape)
+            )
+        last_dim = input_shape[-1][-1]
+        if self.projection_dim is None:
+            kernel_shape = [last_dim, last_dim]
+        else:
+            if self.projection_dim != last_dim:
+                raise ValueError(
+                    "The case where `projection_dim` != last "
+                    "dimension of the inputs is not supported yet, got "
+                    "`projection_dim` {}, and last dimension of input "
+                    "{}".format(self.projection_dim, last_dim)
+                )
+            kernel_shape = [last_dim, self.projection_dim]
+        self.kernel = self.add_weight(
+            "kernel",
+            shape=kernel_shape,
+            initializer=self.kernel_initializer,
+            regularizer=self.kernel_regularizer,
+            dtype=self.dtype,
+            trainable=True,
+        )
+        if self.use_bias:
+            self.bias = self.add_weight(
+                "bias",
+                shape=[last_dim],
+                initializer=self.bias_initializer,
+                regularizer=self.bias_regularizer,
+                dtype=self.dtype,
+                trainable=True,
+            )
+        self.built = True
+
+    def call(self, inputs):
+        if not isinstance(inputs, (tuple, list)) or len(inputs) != 2:
+            raise ValueError(
+                "Inputs to the layer must be a tuple or list of size 2, "
+                "got {}".format(inputs)
+            )
+        x0, x = inputs
+        outputs = x0 * tf.matmul(x, self.kernel) + x
+        if self.use_bias:
+            outputs = tf.add(outputs, self.bias)
+        return outputs
+
+    def get_config(self):
```

Comment:
Please add a test for serialization. Similar to https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/tlu_test.py#L49

Reply:
Done.